cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillController.java

```diff
@@ -484,7 +475,118 @@ private void deletePVCs() {
     return resultFuture;
   }
 
+  /**
+   * Deletes the job controlled by this controller asynchronously.
+   *
+   */
+  private void deleteJob() {
+    apiClient.setDebugging(true);
+    BatchV1Api batchApi = new BatchV1Api(apiClient);
+
+    String name = meta.getName();
+    try {
+      batchApi.deleteNamespacedJobAsync(name, kubeNamespace, null, null, null, null, null,
+                                        new V1DeleteOptions(), new ApiCallbackAdapter<V1Status>() {
+          @Override
+          public void onFailure(ApiException e, int statusCode, Map<String, List<String>> responseHeaders) {
+            // Ignore the failure if the job is already deleted
+            if (statusCode == 404) {
+              LOG.warn("Failed to delete job {}. Job not found.", name, e);
+            }
+          }
+
+          @Override
+          public void onSuccess(V1Status v1Status, int statusCode, Map<String, List<String>> responseHeaders) {
+            deletePods(name);
```

Comment:
Is this needed? The Job should be the parent of the pods it created, in which on the Job deletion, all the pods should get GC by k8s automatically.

Reply:
yea, this is being handled by new k8s client.

Reply:
yes fixed to rely on k8s GC to clean up dependents